qpc/report/summary.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+#
+"""ReportSummaryCommand is used to show summary report information."""
+
+from __future__ import print_function
+import sys
+from requests import codes
+from qpc.utils import pretty_print
+from qpc.clicommand import CliCommand
+import qpc.report as report
+from qpc.request import GET
+from qpc.translation import _
+import qpc.messages as messages
+from qpc.utils import (validate_write_file,
+                       write_file)
+
+
+# pylint: disable=too-few-public-methods
+class ReportSummaryCommand(CliCommand):
+    """Defines the show command.
+
+    This command is for showing the summary report.
+    """
+
+    SUBCOMMAND = report.SUBCOMMAND
+    ACTION = report.SUMMARY
+    SUMMARY_URI = report.REPORT_URI
+
+    def __init__(self, subparsers):
+        """Create command."""
+        # pylint: disable=no-member
+        CliCommand.__init__(self, self.SUBCOMMAND, self.ACTION,
+                            subparsers.add_parser(self.ACTION), GET,
+                            self.SUMMARY_URI, [codes.ok])
+        self.parser.add_argument('--id', dest='id', metavar='ID',
+                                 help=_(messages.REPORT_SUMMARY_ID_HELP),
+                                 required=True)
+
+        group = self.parser.add_mutually_exclusive_group(required=True)
+        group.add_argument('--json', dest='output_json', action='store_true',
+                           help=_(messages.REPORT_SUMMARY_OUTPUT_JSON_HELP))
+        group.add_argument('--csv', dest='output_csv', action='store_true',
+                           help=_(messages.REPORT_SUMMARY_OUTPUT_CSV_HELP))
+
+        self.parser.add_argument('--output-file', dest='path', metavar='PATH',
+                                 help=_(messages.REPORT_SUMMARY_PATH_HELP),
+                                 required=True)
+
+    def _validate_args(self):
+        CliCommand._validate_args(self)
+        if self.args.id:
```

Comment:
Need to take Scan ID and retrieve fact collection. This is a simpler flow for users.